astropy/stats/lombscargle/_lombscargle.py

```diff
@@ -0,0 +1,255 @@
+"""Main Lomb-Scargle Implementation"""
```

Comment:
Usually in astropy, we do not have hidden files, though we do use the underscore for indicating variables that are not public. By analogy of `time` and `units`, perhaps call this file `core.py`?